bootstrap/tracing.go

```diff
@@ -0,0 +1,155 @@
+	if !ok {
+		label = "job"
+	}
+
+	return label
+}
+
+// startTracingDatadog sets up tracing based on the config values. It uses opentracing as an
+// abstraction so the agent can support multiple libraries if needbe.
+func (b *Bootstrap) startTracingDatadog(ctx context.Context) (tracetools.Span, context.Context, stopper) {
+	resourceName := b.OrganizationSlug + "/" + b.PipelineSlug + "/" + b.getLabel()
+	opts := []tracer.StartOption{
+		tracer.WithServiceName("buildkite_agent"),
+		tracer.WithSampler(tracer.NewAllSampler()),
+		tracer.WithAnalytics(true),
+		tracer.WithGlobalTag(ddext.ResourceName, resourceName),
+	}
+
+	tags := Merge(GenericTracingExtras(b, *b.shell.Env), DDTracingExtras())
+	opts = slices.Grow(opts, len(tags))
+	for k, v := range tags {
+		opts = append(opts, tracer.WithGlobalTag(k, v))
+	}
+
+	opentracing.SetGlobalTracer(opentracer.New(opts...))
+
+	wireContext := b.extractTraceCtx()
+
+	span := opentracing.StartSpan("job.run", opentracing.ChildOf(wireContext))
+	ctx = opentracing.ContextWithSpan(ctx, span)
+
+	return tracetools.NewOpenTracingSpan(span), ctx, tracer.Stop
+}
+
+func GenericTracingExtras(b *Bootstrap, env env.Environment) map[string]any {
+	buildID, _ := env.Get("BUILDKITE_BUILD_ID")
+	buildNumber, _ := env.Get("BUILDKITE_BUILD_NUMBER")
+	buildURL, _ := env.Get("BUILDKITE_BUILD_URL")
+	jobURL := fmt.Sprintf("%s#%s", buildURL, b.JobID)
+	source, _ := env.Get("BUILDKITE_SOURCE")
+
+	retry := 0
+	if attemptStr, has := env.Get("BUILDKITE_RETRY_COUNT"); has {
+		if parsedRetry, err := strconv.Atoi(attemptStr); err == nil {
+			retry = parsedRetry
+		}
+	}
+
+	parallel := 0
+	if parallelStr, has := env.Get("BUILDKITE_PARALLEL_JOB"); has {
+		if parsedParallel, err := strconv.Atoi(parallelStr); err == nil {
+			parallel = parsedParallel
+		}
+	}
+
+	rebuiltFromID, has := env.Get("BUILDKITE_REBUILT_FROM_BUILD_NUMBER")
+	if !has || rebuiltFromID == "" {
+		rebuiltFromID = "n/a"
+	}
+
+	triggeredFromID, has := env.Get("BUILDKITE_TRIGGERED_FROM_BUILD_ID")
+	if !has || triggeredFromID == "" {
+		triggeredFromID = "n/a"
+	}
+
+	return map[string]any{
+		"buildkite.agent":             b.AgentName,
+		"buildkite.version":           agent.Version(),
+		"buildkite.queue":             b.Queue,
+		"buildkite.org":               b.OrganizationSlug,
+		"buildkite.pipeline":          b.PipelineSlug,
+		"buildkite.branch":            b.Branch,
+		"buildkite.job_id":            b.JobID,
+		"buildkite.job_url":           jobURL,
+		"buildkite.build_id":          buildID,
+		"buildkite.build_number":      buildNumber,
+		"buildkite.build_url":         buildURL,
+		"buildkite.source":            source,
+		"buildkite.retry":             retry,
+		"buildkite.parallel":          parallel,
+		"buildkite.rebuilt_from_id":   rebuiltFromID,
+		"buildkite.triggered_from_id": triggeredFromID,
+	}
+}
+
+func DDTracingExtras() map[string]any {
+	return map[string]any{
+		ddext.AnalyticsEvent:   true,
```

Comment:
This was previously set on the root span, not as a global tag. I imagine we need to keep that behaviour?

EDIT: Oh, I see we were adding this tag manually on every span in StartSpanFromContext. I guess it makes sense to add it globally.